openedx/edx-enterprise
# feat: adds attribute to hide course prices when zero

### Description
This adds a new attribute `hide_course_price_when_zero` to the EnterpriseCustomer model, which will hide the pricing information from the enrollment page when the final price of a premium course mode is Zero.

### Testing instructions

1. Setup a working devstack setup for edx-enterprise. Required services: LMS, ECommerce, Discovery and Enterprise Catalog (this needs to be setup separately)
2. Check out the PR branch
3. OPTIONAL: Install edx-enterprise in your LMS container if not already installed. 
    ```
    # From devstack dir
    make dev.shell.lms
    pip install -e /edx/src/edx-enterprise
    ```
4. Apply the migrations
    ```
    # From devstack dir
    make dev.shell.lms
    ./manage.py lms migrate
    ```
5. Go to [enterprise customer admin page](http://localhost:18000/admin/enterprise/enterprisecustomer/) copy the UUID of the test enterprise
6. Go to [enterprise catalog admin page](http://localhost:18000/admin/enterprise/enterprisecustomercatalog/) and copy the UUID of the catalog attached to test enterprise
7. Go to [discovery search page](http://localhost:18381/api/v1/search/all/) and copy the `key` of one of the courses included in the enterprise catalog. If you are using the "All course runs" catalog used in the devstack setup, then it can be any course's `key` which has a paid seat.
8. Visit the URL `http://localhost:18000/enterprise/<enterprise_uuid>/course/<course_key>/enroll/`, eg., http://localhost:18000/enterprise/753e5f69-ef92-42ab-ba07-9326d76603e8/course/edX%2BP315/enroll/. This should show the course details along with the price of the course. ![Unchecked](https://github.com/openedx/edx-enterprise/assets/383862/4c40b31e-abe7-4a3b-8925-3d16d8486439)
9. Go to the Ecommerce `/courses/` page and add an **Enterprise Offer** using the enterprise and catalog UUIDs and set the discount percent to 100% (this will mark the final price to zero). Note: Use `006NFJHRVSD683NDI8` for the Salesforce opportunity ID for testing.
10. Now refresh the enrollment page, it should show the final price as zero. ![Unchecked_zero](https://github.com/openedx/edx-enterprise/assets/383862/3bcde3f0-930c-4b25-99bd-e3e2c6193770) ![Unchecked_zero_modal](https://github.com/openedx/edx-enterprise/assets/383862/e5bd09f6-a258-48f2-ab17-e9819ba3fec3)
11. Go to the [enterprise customer admin page](http://localhost:18000/admin/enterprise/enterprisecustomer/) and edit the test enterprise. Check the **Hide course price when zero** flag and save the customer.
12. Refresh the enrollment page. The pricing information would have disappeared. ![image](https://github.com/openedx/edx-enterprise/assets/383862/858be4fb-96e1-4d52-8c7a-1e46c62bd051) 
![image](https://github.com/openedx/edx-enterprise/assets/383862/4a84d589-86ba-4105-95dd-2d9595398152)

### Merge checklist
    - `base.in` if needed in production but edx-platform doesn't install it
    - `test-master.in` if edx-platform pins it, with a matching version
    - `make upgrade && make requirements` have been run to regenerate requirements
    - Checkout the [Database Migration](https://openedx.atlassian.net/wiki/spaces/AC/pages/23003228/Everything+About+Database+Migrations) Confluence page for helpful tips on creating migrations.
    - *Note*: This **must** be run if you modified any models.
      - It may or may not make a migration depending on exactly what you modified, but it should still be run.
    - This should be run from either a venv with all the lms/edx-enterprise requirements installed or if you checked out edx-enterprise into the src directory used by lms, you can run this command through an lms shell.
        - It would be `./manage.py lms makemigrations` in the shell.

### Post merge
    - *Note*: Assets will be added automatically. You just need to provide a tag (should match your version number) and title and description.
    - Each step in the release build has a condition flag that checks if the rest of the steps are done and if so will deploy to PyPi.
    (so basically once your build finishes, after maybe a minute you should see the new version in PyPi automatically (on refresh))
    - This **must** be done after the version is visible in PyPi as `make upgrade` in edx-platform will look for the latest version in PyPi.
    - Note: the edx-enterprise constraint in edx-platform **must** also be bumped to the latest version in PyPi.
